PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -10,6 +10,56 @@
 UINT8    mCurrentBoot;
 VOID    *mEntryStack;
 
+/**
+  Callback function to get and extend digest for OsLoader.
+
+  @param[in]  CbInfo    Component Call Back Info
+
+**/
+VOID
+OsLoaderCallback (
+  IN  COMPONENT_CALLBACK_INFO   *CbInfo
+  )
+{
+  UINT8                      DigestHash[HASH_DIGEST_MAX];
+  RETURN_STATUS              Status;
+  LOADER_PLATFORM_INFO      *LoaderPlatformInfo;
+  HASH_ALG_TYPE              MbHashType;
+  TPMI_ALG_HASH              MbTmpAlgHash;
+  UINT8                     *HashPtr;
+
+  //Convert Measured boot Hash Mask to HASH_ALG_TYPE (CryptoLib)
+  MbHashType   = GetCryptoHashAlg(PcdGet32(PcdMeasuredBootHashMask));
+
+  //Convert Measured boot Hash Mask to TPMI_ALG_HASH (TPM ALG ID)
+  MbTmpAlgHash = (TPMI_ALG_HASH) GetTpmHashAlg(PcdGet32(PcdMeasuredBootHashMask));
+
+  LoaderPlatformInfo = (LOADER_PLATFORM_INFO *)GetLoaderPlatformInfoPtr();

```

Comment:
Create abstraction layer so that it does not need LAODER_PLATFORM_INFO.   Then the code can be common between Stage and Payload.

Reply:
BootloaderCore has MEASURED_BOOT_ENABLED() support and it currently getting used in stage components. 
Check for measured boot using LAODER_PLATFORM_INFO would be limited to the Osloader only.

Reply:
Please use GetFeatureCfg() instead of LOADER_PLATFORM_INFO.

Reply:
Thanks Aiden. I will check.